src/main/cpp/Monopoly.cpp

```diff
@@ -240,13 +280,243 @@ Monopoly::print(string s, bool wait, bool reset) {
 	
 }
 
+void Monopoly::drawChanceCard() {
+	int cardNum = rand() % 16;
+	shared_ptr<Player> player = getCurrentPlayer();
+	shared_ptr<Player> owner;
+	int originalSpot = player->getLocationNum();
+	BoardTile *newSpot;
```

Comment:
Suggest using shared_ptr here as well. You may have to update some function definitions/return types tho.

Reply:
I'm not sure if I'll be able to. In lines 153 and 189 in Monopoly.cpp, I type cast a BoardTile pointer as a HouseTile pointer. If I convert the tile classes to shared_ptr, I won't be able to do type casting.

Reply:
You should be able to type cast a shared_ptr. Check out this thread:

https://stackoverflow.com/questions/1358143/downcasting-shared-ptrbase-to-shared-ptrderived